File: ws/santorum/u2/ws12Inheritance/FarmSystem/src/main/java/ec/edu/espe/farmsystem/model/Cage.java
package ec.edu.espe.farmsystem.model;

/**
 *
 * @author Thais Santórum
 */
public class Cage {
    private int id;
    private String description;
    private Location location;
    private int type; // 1 coop / 2 stable / 3 pens

    @Override
    public String toString() {
        return "Cage{" + "id=" + getId() +
                ", description=" + getDescription() +
                ", location=" + getLocation() +
                ", type=" + getType()
                + '}';
    }

    public Cage(int id, String description, int type, Location location) {
        this.id = id;
        this.description = description;
        this.location = location;
        this.type = type;
    }

    /**
     * @return the id
     */
    public int getId() {
        return id;
    }

    /**
     * @param id the id to set
     */
    public void setId(int id) {
        this.id = id;
    }

    /**
     * @return the description
     */
    public String getDescription() {
        return description;
    }

    /**
     * @param description the description to set
     */
    public void setDescription(String description) {
        this.description = description;
    }

    /**
     * @return the location
     */
    public Location getLocation() {
        return location;
    }

    /**
     * @param location the location to set
     */
    public void setLocation(Location location) {
        this.location = location;
    }

    /**
     * @return the type
     */
    public int getType() {
        return type;
    }

    /**
     * @param type the type to set
     */
    public void setType(int type) {
        this.type = type;
    }

    
    

 
    
    
}

File: hw/calle/u2/hw19Inheritance/FarmsystemJavaScript/modules/Cage.js
import { Location } from './Location.js';

export class Cage {
    #id;
    #description;
    #type;
    #location;

    constructor(id, description, type, location) {
        this.#id = id;
        this.#description = description;
        this.#type = type;
        this.#location = location;
    }

    toString() {
        return `Cage{id=${this.#id}, description=${this.#description}, type=${this.#type}, location=${this.#location.toString()}}`;
    }

    getId() {
        return this.#id;
    }

    setId(id) {
        this.#id = id;
    }

    getDescription() {
        return this.#description;
    }

    setDescription(description) {
        this.#description = description;
    }

    getType() {
        return this.#type;
    }

    setType(type) {
        this.#type = type;
    }

    getLocation() {
        return this.#location;
    }

    setLocation(location) {
        this.#location = location;
    }
}